packages/core/docs/commercetools/composables/use-forgot-password.md

```diff
@@ -0,0 +1,184 @@
+# `useForgotPassword`
+
+## Features
+
+`useForgotPassword` composable can be used to:
+
+* generate reset password token
+* set new password using token
+
+## API
+
+- `request` - function to generate reset password token. When invoked, it requests data from the API and populates the `token` property. This method accepts a single `params` object with the following properties:
+
+  - `params: ResetPasswordParams`
+
+  - `customQuery?: CustomQuery`
+
+```typescript
+interface ResetPasswordParams {
+    email: string;
+}
+
+type CustomQuery = {
+  customerCreatePasswordResetToken: string
+}
+```
+
+- `setNew` - function to set new user password after `request`. When invoked, it requests data from the API and populates the `result` object. This method accepts a single `params` object with the following properties:
+
+  - `params: SetNewPasswordParams`
+  
+  - `customQuery?: CustomQuery`
+
+```typescript
+interface SetNewPasswordParams {
+    tokenValue: string;
+    newPassword: string;
+}
+
+type CustomQuery = {
+  customerResetPassword: string
+}
+```
+
+- `token: string` - reactive data string containing the reset token.
+
+- `loading: boolean` - reactive object containing information about loading state of `setNew` and `request` methods.
+
+- `error: UseForgotPasswordErrors` - reactive object containing the error message, if `setNew` or `request` failed for any reason.
+
+```ts
+interface UseForgotPasswordErrors {
+  request: Error;
+  setNew: Error;
+}
+```
+
+## Getters
+
+- `getResetPasswordToken` - returns generated reset password token.
+
+- `isPasswordChanged` - returns a boolean value of a password set status.
+
+```ts
+interface ForgotPasswordGetters<ForgotPasswordResult> {
+  getResetPasswordToken: (result: ForgotPasswordResult) => string
+  isPasswordChanged: (result: ForgotPasswordResult) => boolean
+}
+
+interface ForgotPasswordResult {
+  resetPasswordResult: CreatePasswordResetTokenResponse;
+  setNewPasswordResult: ResetPasswordResponse;
+}
+
+type CreatePasswordResetTokenResponse = QueryResponse<'customerCreatePasswordResetToken', CustomerPasswordToken>;
+
+type ResetPasswordResponse = QueryResponse<'customerResetPassword', Customer>;
+
+type CustomerPasswordToken = {
+  customerId: Scalars["String"];
+  expiresAt: Scalars["DateTime"];
+  value: Scalars["String"];
+  id: Scalars["String"];
+  version: Scalars["Long"];
+  createdAt: Scalars["DateTime"];
+  lastModifiedAt: Scalars["DateTime"];
+  createdBy?: Maybe<Initiator>;
+  lastModifiedBy?: Maybe<Initiator>;
+}
```

Comment:
This should come from API reference. Please ask @andrzejewsky to generate new GraphQL types and generate new API reference. Then link to instead of pasting whole interface

Reply:
Created Jira issue for that as this recommendation is out of the forgot password and will be needed in other places as well.

Reply:
@filipsobol completely right, but let's do this in a different task, because it sound like other scope